docs/istio.md

```diff
@@ -0,0 +1,204 @@
+# SMI Metrics with Istio
+
+The SMI Metrics Extension APIService makes use of prometheus metrics exposed by Istio to resopond to metric queries like p90, request counts, etc.
+
+There is a configuration change required on the Istio metrics for SMI Metrics to work, as the default metrics dosen't have enough information to relate thosoe metrics with that of Kubernetes resources.
+
+SMI Metrics makes use of the following Metric Instances in Istio:
+
+- `istio_request_duration_seconds_bucket`
+- `istio_requests_total`
+
+As SMI Metrics allows users to query for metrics based on the resource `kind` i.e pod, deployment, etc. These metrics should have corresponding labels that provide this information.
+
+The labels required by SMI Metrics on the above metrics are:
+
+- `source_owner`, `destination_owner` : To identify the workload types that these metrics belong to i.e Deployment, Daemonset, Job, etc.
+- `source_uid`, `destination_uid` : To identify the pod instance that these metrics belong to.
+- `source_workload_namespace`,  `destination_workload_namespace` : To identify the namespace that a particular metric belongs to.
+
+The first four labels are not present in the default installation of Istio. So, For SMI to work with Istio, these four labels have to be added manually to the above mentioned Metric Instances.
+
+This can be done by adding those labels to the `dimensions` field of the corresponding Metric Instance definitions and the `label_names` field of Handler definitions.
+
+```diff
+diff --git install/kubernetes/istio-demo-auth.yaml install/kubernetes/istio-demo-auth.yaml
```

Comment:
Is it possible to do this with the helm install instead? That'd remove a lot of this diff template stuff.

Reply:
Adding these cr's to helm installation should work! but that would override any unique labels that they have, that are not present in our cr's!